--- origsrc/httpd-2.2.21/support/apxs.in	2011-09-21 13:19:52.789478800 -0500
+++ src/httpd-2.2.21/support/apxs.in	2011-09-21 13:21:05.754652200 -0500
@@ -329,14 +329,14 @@ if ($opt_q) {
 
 my $apr_config = get_vars("APR_CONFIG");
 
-if (! -x "$apr_config") {
+if (($opt_c or $opt_i) and ! -x "$apr_config") {
     error("$apr_config not found!");
     exit(1);
 }
 
 my $apu_config = get_vars("APU_CONFIG");
 
-if (! -x "$apu_config") {
+if (($opt_c or $opt_i) and ! -x "$apu_config") {
     error("$apu_config not found!");
     exit(1);
 }
