--- origsrc/httpd-2.2.9/configure.in	2008-06-09 11:04:46.000000000 -0500
+++ src/httpd-2.2.9/configure.in	2008-07-31 11:51:07.921875000 -0500
@@ -235,7 +235,7 @@
       LIBTOOL="$my_libtool \$(LTFLAGS)"
       libtoolversion=`$my_libtool --version`
       case $libtoolversion in
-          *1.[[45]]*)
+          *1.[[45]]*|*2.2*)
               SH_LIBTOOL='$(LIBTOOL)'
               SHLTCFLAGS="-prefer-pic"
               LTCFLAGS="-prefer-non-pic -static"
