--- origsrc/httpd-2.2.9/docs/cgi-examples/printenv	2004-12-10 23:52:29.000000000 -0600
+++ src/httpd-2.2.9/docs/cgi-examples/printenv	2008-07-31 11:16:51.093750000 -0500
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 ##
 ##  printenv -- demo CGI program which just prints its environment
 ##
--- origsrc/httpd-2.2.9/docs/conf/httpd.conf.in	2007-12-21 05:43:00.000000000 -0600
+++ src/httpd-2.2.9/docs/conf/httpd.conf.in	2008-07-31 11:16:03.187500000 -0500
@@ -63,8 +63,8 @@
 # It is usually good practice to create a dedicated user and group for
 # running httpd, as with most system services.
 #
-User daemon
-Group daemon
+#User daemon
+#Group daemon
 
 </IfModule>
 </IfModule>
@@ -399,6 +399,10 @@
 
 # Secure (SSL/TLS) connections
 #Include @rel_sysconfdir@/extra/httpd-ssl.conf
+
+# Provide a way for additional packages to add configuration fragments
+Include @rel_sysconfdir@/conf.d/*.conf
+
 #
 # Note: The following must must be present to support
 #       starting without SSL on platforms with no /dev/random equivalent
