--- origsrc/httpd-2.4.12/modules/ssl/config.m4	2013-11-15 11:06:18.000000000 -0600
+++ src/httpd-2.4.12/modules/ssl/config.m4	2015-05-21 15:42:07.999691600 -0500
@@ -42,7 +42,7 @@ APACHE_MODULE(ssl, [SSL/TLS support (mod
         if test "x$enable_ssl" = "xshared"; then
            # The only symbol which needs to be exported is the module
            # structure, so ask libtool to hide everything else:
-           APR_ADDTO(MOD_SSL_LDADD, [-export-symbols-regex ssl_module])
+           APR_ADDTO(MOD_SSL_LDADD, [-lssl -lcrypto -export-symbols-regex ssl_module])
         fi
     else
         enable_ssl=no
